datumaro/components/converter.py

```diff
@@ -34,8 +36,22 @@ def convert(cls, extractor, save_dir, **options):
         return converter.apply()
 
     @classmethod
+    @error_rollback
     def patch(cls, dataset, patch, save_dir, **options):
-        return cls.convert(dataset, save_dir, **options)
+        # Is not any better in performance than just writing a dataset,
+        # but in case of patching (i.e. writing to the previous location),
+        # it allows to avoid many problems with removing and replacing
+        # existing files. Surely, this approach also has problems with
+        # removal of the given directory.
+        tmpdir = save_dir + '.tmp' + uuid.uuid4().hex
+        os.makedirs(tmpdir, exist_ok=False)
```

Comment:
```suggestion
        os.mkdir(tmpdir)
```

This is a rare occasion on which creating parent directories is unnecessary.

Reply:
In normal cases yes, but if user removed something... I don't think this question really should be considered here.

Reply:
I guess that's a fair point.